app/controllers/users/webauthn_setup_controller.rb

```diff
@@ -21,6 +21,9 @@ def new
       )
       result = form.submit(new_params)
       @platform_authenticator = form.platform_authenticator?
+      if ft_unlock_setup_successful_on_account_setup?
```

Comment:
This seems like it'd be a good candidate for a `before_action`, which is typically how we'd isolate some behavior that may redirect under specific conditions.

The nice thing about it is it prevents any other code here from being run, which I don't think would happen with the current code (e.g. logging, saving challenge in session, etc.)

>If a "before" filter renders or redirects, the action will not run

https://guides.rubyonrails.org/action_controller_overview.html#filters